### PR TITLE
Scheduler dataset

### PR DESCRIPTION
**Pull request by anwar6953**
@anew's implementation of ScheduleStore, but rebased with latest code of of the integration branch (feature/CDAP-7593-scheduler), and addressing the comments from https://github.com/caskdata/cdap/pull/8685.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+  /**
+   * Add one or more schedules to the store.
+   *
+   * @param schedules the schedules to add
+   *
+   * @throws AlreadyExistsException if one of the schedules already exists
+   */
+  public void addSchedules(ProgramSchedule... schedules) throws AlreadyExistsException {
+    addSchedules(Lists.newArrayList(schedules));
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java]: Use `Arrays.asList` instead.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    }
+  }
+
+  /**
+   * Removes one or more schedules from the store. Succeeds whether the schedules exist or not.
+   *
+   * @param scheduleIds the schedules to delete
+   */
+  public void deleteSchedules(ScheduleId ... scheduleIds) {
+    deleteSchedules(Lists.newArrayList(scheduleIds));
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/ProgramScheduleStoreDataset.java]: Use `Arrays.asList`.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/Schedulers.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.schedule.store;
+
+import co.cask.cdap.proto.id.DatasetId;
+
+/**
+ * Common static methods to handle trigger keys.
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/Schedulers.java]: Update javadoc. Common static methods for scheduler.

**Comment by anwar6953** [comment]
Replaced by https://github.com/caskdata/cdap/pull/8695

**Pull request closed without merging**